### PR TITLE
Make bevy_render an optional dependency of bevy_scene

### PR DESCRIPTION
**Pull request by Neo-Zhixing**
# Objective

bevy-scene does not have a reason to depend on bevy-render except to include the `Visibility` and `ComputedVisibility` components. Including that in the dependency chain is unnecessary for people not using `bevy_render`.

Also fixed a problem where compilation fails when the `serialize` feature was not enabled.

## Solution

This was added in #5335 to address some of the problems caused by #5310.

Imo the user just always have to remember to include `VisibilityBundle` when they spawn `SceneBundle` or `DynamicSceneBundle`, but that will be a breaking change. This PR makes `bevy_render` an optional dependency of `bevy_scene` instead to respect the existing behavior.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_scene/Cargo.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ license = "MIT OR Apache-2.0"
 keywords = ["bevy"]
 
 [features]
-default = ["serialize"]
+default = ["serialize", "bevy_render"]
```
Comment (james7132) [inline crates/bevy_scene/Cargo.toml:12]: I'm questioning if this should be a default feature. The motivations for this PR seems to suggest it shouldn't be, and should be enabled in `bevy_internal` or `bevy` proper instead.

**Review by james7132 — CHANGES_REQUESTED** [review]

File: crates/bevy_scene/Cargo.toml (hunk 1 above)
Comment (Neo-Zhixing) [reply crates/bevy_scene/Cargo.toml]: I've made the change as requested.

**Review by james7132 — APPROVED** [review]

**Comment by mockersf** [comment]
I don't think this makes it possible to enable the `bevy_scene` features without depending on `bevy_render`

**Comment by Neo-Zhixing** [comment]
@mockersf  I've made it so that bevy_scene/bevy_render only gets enabled when bevy_render was enabled.

**Pull request merged**